java-backend/src/main/java/org/kframework/backend/java/symbolic/FastRuleMatcher.java

```diff
@@ -231,12 +233,15 @@ private BitSet matchAndLog(Term subject, Term pattern, BitSet ruleMask,
         BitSet result;
         try {
             result = match(subject, pattern, ruleMask, path, logFailures);
+        } catch (KEMException e) {
+            addDetailedStackFrame(e, subject, pattern);
+            throw e;
             // DISABLE EXCEPTION CHECKSTYLE
         } catch (RuntimeException | AssertionError | StackOverflowError e) {
             // ENABLE EXCEPTION CHECKSTYLE
-            System.err.format("\nException during matching rule pattern:\n\t%s\nSubject:\n%s\nPattern:\n%s\n",
-                    e.getMessage(), getLogString(subject), getLogString(pattern));
-            throw e;
+            KEMException newExc = KEMException.criticalError("", e);
```

Comment:
There should be a message for this exception.

Reply:
it's the message of original exception. K-specific stack frames give more info. Tested, looks best this way.

Reply:
If you don't pass --debug, you're going to get an error with no text. That's not really great.

Reply:
@dwightguth Nope. Nice exception message will be present with and without --debug.
With --debug:
```
org.kframework.utils.errorsystem.KEMException: [Error] Critical:  (unexpected subject class at matching: class org.kframework.backend.java.kil.KList)
  while matching rule pattern:
...
	at org.kframework.utils.errorsystem.KEMException.create(KEMException.java:108)
...
```

Without --debug:
```
[Error] Critical:  (unexpected subject class at matching: class org.kframework.backend.java.kil.KList)
  while matching rule pattern:
...
```

Reply:
Ah, I didn't realize we were printing the exception message as part of the kexception now. It's possible that might generate some output that contains stuff the user doesn't care about, but we can address that if we find an example of that. Alright, I'm okay with this for now then.